File: app/src/api/product-catalog/data-transfer-objects/requests/pad-series-update-request.ts
import { PadMaterial } from '../pad-material';
import { PadTexture } from '../pad-texture';
import { PolisherType } from '../polisher-type';
import { PadCreateOrUpdate } from './pad-create-or-update';
import { PadSizeCreateOrUpdate } from './pad-size-create-or-update';

export interface PadSeriesUpdateRequest {
    id: string;
    name: string;
    brandId: string;
    polisherTypes: PolisherType[];
    sizes: PadSizeCreateOrUpdate[];
    pads: PadCreateOrUpdate[];
}

File: api/src/Api/ProductCatalog/DataTransferObjects/Requests/PadSeriesUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using DetailingArsenal.Domain.ProductCatalog;

namespace DetailingArsenal.Api.ProductCatalog {
    public class PadSeriesUpdateRequest : IDataTransferObject {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public Guid BrandId { get; set; }
        public string Texture { get; set; } = null!;
        public string Material { get; set; } = null!;
        public List<string> PolisherTypes { get; set; } = new();
        public List<PadSizeCreateOrUpdate> Sizes { get; set; } = new();
        public List<PadColorCreateOrUpdate> Colors { get; set; } = new();
    }
}